src/protagonist/Engine/Ingest/Image/Appetiser/AppetiserClient.cs

```diff
@@ -183,11 +186,11 @@ private static void UpdateImageDimensions(Asset asset, AppetiserResponseModel re
         asset.Width = responseModel.Width;
     }
 
-    private async Task ProcessOriginImage(IngestionContext context, ImageProcessorFlags processorFlags)
+    private async Task ProcessOriginImage(IngestionContext context, ImageProcessorFlags processorFlags, AssetId modifiedAssetId)

```

Comment:
I don't think we want to change the key in S3 if it contains a bracket character - or if we do it should happen when all S3 keys are generated (ie in `IStorageKeyGenerator` impl), rather than just in `AppetiserClient`.

As-is we could end up with S3 keys that are inconsistent for the same object. If I create `9/9/test(1)` then in S3 we could end up with:
* `thumbs-bucket/9/9/test(1)/s.json` - s.json for thumbns
* `thumbs-bucket/9/9/test(1)/100.jpg` - a 100px thumb
* `storage-bucket/9/9/test(1)/cantaloupe/v3/info.json` - cached info.json
* `storage-bucket/9/9/test_1_` - derivative jp2 for use by image-server
* `storage-bucket/9/9/test(1)/original` - source image for use by image-server (if `use-original`)

The root issue is how linux handles `(` and `)` in filenames - we can either escape/replace these when writing to disk or update where they are stored in S3. cc @tomcrane for thoughts

Reply:
I agree that the names in S3 should always be consistent.
Asset identifiers, with this restriction, will always contain S3-key-safe characters. It's only when a file gets moved to a Unix filesystem for use by an image server that we have an issue, is that right?

Orchestrator does this - but does Cantaloupe also do it (if we configured it) for its own local disk cache? may be worth seeing what Cantaloupe does in that scenario.

Is it the case that `(` and `)` are the only chars we'd want in our whitelist that the filesystem doesn't like? if so we could have a special escape sequence for them `__open_p__` `__close_p__` or something.

We have a lot of identifiers in Wellcome that contain invalid chars - but they won't be images that need orchestrating - they'll be MP3s and similar.

Reply:
> It's only when a file gets moved to a Unix filesystem for use by an image server that we have an issue, is that right?

The original issue was when file moved to unix filesystem for use by appetiser, rather than image-server. Orchestrator doesn't change or escape characters when copying to scratch disk so we may run into the issue on Orchestration too - will need to test to verify.